### PR TITLE
Update TESTING.asciidoc for new :qa:os test style

### PR DESCRIPTION
**Pull request by williamrandolph**
The TESTING.asciidoc document had gotten out of date due to some new and
wonderful changes in our vagrant testing code. I've taken a first cut at
some updates here.

Ryan and co. have done a lot to move coordinating code from various little scripts into Gradle tasks and plugins, so we have lost some of the "granularity" we used to have.

I'm sure I'll be cleaning this up a little once I see what it looks like in GitHub.

### CODE REVIEW

**Comment by williamrandolph** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by williamrandolph** [comment]
@elasticmachine run elasticsearch-ci/bwc

File: TESTING.asciidoc [hunk 1]
```diff
@@ -313,39 +321,28 @@ a bit more performance out of the process:
...
-Note that if you interrupt gradle in the middle of running these tasks, any boxes started
+If you want a quick test of the tarball and RPM packagings for Centos 7, you
+would run:
++
+-------------------------------------------------------------------------------------------------
+./gradlew :qa:os:centos-7:distroTest.default-rpm :qa:os:centos-7:distroTest.default-linux-archive
+-------------------------------------------------------------------------------------------------
+
+Note that if you interrupt Gradle in the middle of running these tasks, any boxes started
 will remain running and you'll have to stop them manually with `./gradlew stop` or
```
Comment (mark-vieira) [inline TESTING.asciidoc]: This should be `./gradlew --stop`. @rjernst I assume it's the reaper process that goes back and kills these VMs when the daemon is killed?

File: TESTING.asciidoc [hunk 2]
```diff
@@ -438,84 +433,51 @@ that'd consume a ton of ram.
...
-  org.elasticsearch.packaging.test.DefaultWindowsZipTests \
-  org.elasticsearch.packaging.test.OssWindowsZipTests
---------------------------------------------
+-----------------------------------------------------------
+./gradlew :qa:os:fedora-28:distroTest.default-linux-archive \
+  --tests "com.elasticsearch.packaging.test.ArchiveTests"
+-----------------------------------------------------------
 
-or on Windows, from a terminal running as Administrator
+If you need to work on the packaging test framework or run Gradle tests from
```
Comment (mark-vieira) [inline TESTING.asciidoc]: Yeah, this bit is pretty goofy. If this is something we expect folks to do, we should bake this capability into the plugin and provide a task for folks just to spin up a "ready-to-go" VM.

File: TESTING.asciidoc [hunk 3]
```diff
@@ -657,7 +619,7 @@ care.
 == Test coverage analysis
```
Comment (mark-vieira) [inline TESTING.asciidoc]: I'm wonder if we can remove this whole section. Test coverage analysis is not part of our development process in any formal way. This assumes that contributors want some way of analyzing test coverage themselves? I'm not sure just saying "you're on your own" is even worth mentioning.

File: TESTING.asciidoc [hunk 4]
```diff
@@ -313,39 +321,28 @@ a bit more performance out of the process:
 vagrant plugin install vagrant-cachier
 --------------------------------------
 +
-. Validate your installed dependencies:
+
+. You can run all of the OS packaging tests with `./gradlew distroTest`. This
```
Comment (rjernst) [inline TESTING.asciidoc]: This isn't quite true. `distroTest` is all of the newer java based tests, but does not include the legacy bats tests. To run both, the `packagingTest` task must be used.

File: TESTING.asciidoc (hunk 1 above)
Comment (rjernst) [reply TESTING.asciidoc]: Yes the reaper will shut the VMs down.

File: TESTING.asciidoc [hunk 5]
```diff
@@ -438,84 +433,51 @@ that'd consume a ton of ram.
...
-cd $PACKAGING_ARCHIVES
-
-# runs all bats tests
-sudo bats $BATS_TESTS/*.bats
-
-# you can also pass specific test files
-sudo bats $BATS_TESTS/20_tar_package.bats $BATS_TESTS/25_tar_plugins.bats
---------------------------------------------
+Because our packaging tests are capable of testing many combinations of OS (Windows
+or Linux), package type (zip file or RPM), Elasticsearch distribution type (default
```
Comment (rjernst) [inline TESTING.asciidoc]: I would use `e.g.` with these parentheticals and maybe `etc` as well, as zip and RPM are not the only package types.

File: TESTING.asciidoc (hunk 2 above)
Comment (rjernst) [reply TESTING.asciidoc]: This should be possible, but we should give sufficient large warnings/notification that the task is going to leave the VM running, and other guards (eg nothing should depend on such a task, it should only be run directly). A followup issue for discussion/impl details seems fine.

File: TESTING.asciidoc (hunk 3 above)
Comment (rjernst) [reply TESTING.asciidoc]: Let's leave that for a separate issue, since this PR is about packaging test docs.

**Review by rjernst — APPROVED** [review]
Looks good, I have a few minor suggestions.

File: TESTING.asciidoc (hunk 2 above)
Comment (williamrandolph) [reply TESTING.asciidoc]: Cool — I'll remove this section but make a new issue for team discussion. Thanks.

File: TESTING.asciidoc (hunk 4 above)
Comment (williamrandolph) [reply TESTING.asciidoc]: Ah, I didn't know that. Thank you for the correction.

File: TESTING.asciidoc (hunk 2 above)
Comment (williamrandolph) [reply TESTING.asciidoc]: I've written a follow-up ticket for discussion: https://github.com/elastic/elasticsearch/issues/48087

**Comment by williamrandolph** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Comment by williamrandolph** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix

**Pull request merged**